### PR TITLE
Dockerize frontend + Update docker compose + Minor changes (See description)

### PR DESCRIPTION
**Pull request by TheSpaceCuber**
* Standardize to `USER_SERVICE_URL` and `MATCHING_SERVICE_URL` instead of `baseUrl`
* Simplified `configs.js`
* Updated in `LoginPage.js` and `SkeletonSignUpPage.js` for consistency, but if not needed, can delete them since the code is redundant in those files.

* Removed MUI and emotion from `package.json` which is installed alongside MUI usually

* New docker compose file `docker-compose.backend.yml` for only backend services if you want to work on frontend dev only
* Main docker compose `docker-compose.yml` for all backend + frontend

* Update README

If you need to test docker-compose.yml, simply run `docker-compose up` in the root directory and go to localhost:3000

If you need to test docker-compose.backend.yml + frontend separately, simply run `docker-compose -f docker-compose.backend.yml up` and cd into the frontend directory to run `npm start`.

### CODE REVIEW

File: frontend/src/components/SkeletonSignUpPage.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import {
 } from "@mui/material";
```
Comment (ongweisheng) [inline frontend/src/components/SkeletonSignUpPage.js:11]: If don't need this file can just delete? @hiongkaihan @TheSpaceCuber

**Review by ongweisheng — APPROVED** [review]
LGTM!

File: frontend/src/components/SkeletonSignUpPage.js (hunk 1 above)
Comment (hiongkaihan) [reply frontend/src/components/SkeletonSignUpPage.js]: Fixed

**Review by hiongkaihan — APPROVED** [review]
LGTM

**Pull request merged**